src/docs/cookbook/animation/physics-simulation.md

```diff
@@ -243,24 +245,37 @@ Finally, `AnimationController` has an `animateWith()` method that can be given a
 [SpringSimulation][]:
 
 ```dart
-onPanEnd: (details) {
-  _updateAnimation();
+void _runAnimation(Offset pixelsPerSecond, Size size) {
+  _animation = _controller.drive(
+    AlignmentTween(
+      begin: _dragAlignment,
+      end: Alignment.center,
+    ),
+  );
   // Calculate the velocity relative to the unit interval, [0,1],
   // used by the animation controller.
-  var pxPerSecond = details.velocity.pixelsPerSecond;
-  var unitsPerSecondX = pxPerSecond.dx / size.width;
-  var unitsPerSecondY = pxPerSecond.dy / size.height;
-  var unitsPerSecond = Offset(unitsPerSecondX, unitsPerSecondY);
-  var unitVelocity = unitsPerSecond.distance;
+  final unitsPerSecondX = pixelsPerSecond.dx / size.width;
+  final unitsPerSecondY = pixelsPerSecond.dy / size.height;
+  final unitsPerSecond = Offset(unitsPerSecondX, unitsPerSecondY);
+  final unitVelocity = unitsPerSecond.distance;
 
-  var spring = SpringDescription(
+  const spring = SpringDescription(
```

Comment:
It's slightly weird to have a constant declared in a function scope rather than in the scope of the file or Widget class itself. If this is the only way to make the article readable, then it's cool to leave it. If you can somehow tell people to add this constant in a more appropriate place without overly complicating the instructions, though, that'd be a good idea.

Reply:
discussed in the other review, but I think we should leave it here for readability